tracer/test/Datadog.Trace.Tests/Telemetry/Collectors/RedactedErrorLogCollectorTests.cs

```diff
@@ -158,6 +159,195 @@ public void RejectsOverSizedLogs()
+            {
+                StackTrace = ExceptionRedactor.Redact(ex, InstrumentedAssemblies)
+            });
+        }
+
+        var batch = collector.GetLogs()
+                             .Should()
+                             .HaveCount(1)
+                             .And.ContainSingle()
+                             .Subject;
+
+        batch.Should().HaveCount(2);
+        batch[0].Message.Should().Be("This is my message");
+        batch[1].Message.Should().Be("This is my message");
+
+        Exception GetException1()
+        {
+            try
+            {
+                throw new Exception(nameof(GetException1));
+            }
+            catch (Exception ex)
+            {
+                return ex;
+            }
+        }
+
+        Exception GetException2()
+        {
+            try
+            {
+                throw new Exception(nameof(GetException1));
+            }
+            catch (Exception ex)
+            {
+                return ex;
+            }
+        }
+    }
+
+    [Fact]
+    public void WhenDeDupeEnabled_AndSendsBatchInBetween_SendsSingleLogPerBatch()
+    {
+        const string message = "This is my message";
+        var collector = new RedactedErrorLogCollector();
+
+        // Add multiple identical messages
+        for (var i = 0; i < 5; i++)
+        {
+            collector.EnqueueLog(new LogMessageData(message, TelemetryLogLevel.ERROR, DateTimeOffset.UtcNow));
+        }
+
+        var firstBatch = collector.GetLogs();
+
+        // Add multiple identical messages
+        for (var i = 0; i < 5; i++)
+        {
+            collector.EnqueueLog(new LogMessageData(message, TelemetryLogLevel.ERROR, DateTimeOffset.UtcNow));
+        }
+
+        var secondBatch = collector.GetLogs();
+        var thirdBatch = collector.GetLogs();
+
+        firstBatch.Should()
+                  .NotBeNull()
+                  .And.ContainSingle()
+                  .Which.Should()
+                  .ContainSingle()
+                  .Subject.Message.Should()
+                  .Be(message);
+
+        secondBatch.Should()
+                  .NotBeNull()
+                  .And.ContainSingle()
+                  .Which.Should()
+                  .ContainSingle()
+                  .Subject.Message.Should()
+                  .Be(message);
+
+        thirdBatch.Should().BeNull();
+    }
+
+    [Fact]
+    public void WhenDeDupeEnabled_AndOverfills_GroupsAllIntoSingleBatch()
+    {
+        const string message = "This is my message";
+        var collector = new RedactedErrorLogCollector();
+        var messagesToSend = RedactedErrorLogCollector.MaximumQueueSize * 2;
+
+        // Add too many identical messages
+        for (var i = 0; i < messagesToSend; i++)
+        {
+            collector.EnqueueLog(new LogMessageData(message, TelemetryLogLevel.ERROR, DateTimeOffset.UtcNow));
+        }
+
+        var logs = collector.GetLogs();
```

Comment:
Shouldn't you add a second call to GetLogs and verify it's empty?

Reply:
Done